src/ARMJIT_A64/ARMJIT_Compiler.h

```diff
@@ -187,6 +187,9 @@ class Compiler : public Arm64Gen::ARM64XEmitter
     void Comp_RegShiftReg(int op, bool S, Op2& op2, Arm64Gen::ARM64Reg rs);
 
     bool Comp_MemLoadLiteral(int size, bool signExtend, int rd, u32 addr);
+
+    static void UpdateMode(ARM* arm, u32 oldmode, u32 newmode);
```

Comment:
make this just a normal function in ARMJIT_Compiler.cpp. Name it something like TrampolineARMUpdateMode